### PR TITLE
Update titan arch files and README

### PR DESCRIPTION
**Pull request by helen-23**
#### Description

Added a 'legacy' sub-directory and moved the old architecture file in there. Also added three older variants from the Titan repository into the 'legacy' directory to keep as record (upgraded files using upgrade_arch.py). Updated README accordingly.

#### Related Issue
Titan Architecture README is out of date #1412

#### Motivation and Context
Reorganizing the files and updating the README are necessary to keep the files easily accessible.

#### How Has This Been Tested?
No code change; no testing required.

#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: vtr_flow/arch/titan/README.rst [hunk 1]
```diff
@@ -24,46 +26,72 @@ CAD evaluation).
...
 stratixiv_arch.timing.xml:
-    A capture of Altera's Stratix IV FPGA architecture. It makes some 
-    relatively minor architectural approximations to be compatible with
-    VPR.  It includes a timing model which has been calibrated to the
-    Stratix IV timing model included in Altera's Quartus II CAD tools.
+    An enhanced capture of Altera's Stratix IV FPGA architecture. It makes some 
+    relatively minor architectural approximations to be compatible with VPR. It
+    includes a timing model which has been calibrated to the Stratix IV timing
+    model included in Altera's Quartus II CAD tools.
+
```
Comment (vaughnbetz) [inline vtr_flow/arch/titan/README.rst:35]: Compared to the original (legacy) capture of Stratix IV, this architecture better captures the routing architecture by modeling the hierarchical connections between the L16 and L4 networks, and the custom switch patterns between them.  The L4 wires have 12-input muxes driving them, and the L16 wires have 40:1 muxes driving them, roughly matching the Stratix IV mux sizes from the information available.

This is the architecture used to generate results in the VTR 8.0 TRETS paper, and corresponds to architecture E in Table 3 in that paper.

Comment (vaughnbetz) [reply vtr_flow/arch/titan/README.rst:35]: (Please add comment above to text).

File: vtr_flow/arch/titan/README.rst [hunk 2]
```diff
@@ -24,46 +26,72 @@ CAD evaluation).
...
+For more details about the Altera's Stratix IV FPGA architecture capture variants,
+see Section 5.4 of the paper, "VTR 8: High Performance CAD and Customizable FPGA Architecture Modelling": 
+<https://www.eecg.utoronto.ca/~kmurray/vtr/vtr8_trets.pdf>
+
+**legacy subdirectory:**
 
-    Use this architecture file, unless you have specific reasons to use 
-    stratixiv_arch.detailed.xml or stratixiv_arch.simple.xml.
+    stratixiv_arch.timing.legacy.xml:
+        The old capture of Altera's Stratix IV FPGA architecture.
```
Comment (vaughnbetz) [inline vtr_flow/arch/titan/README.rst:62]: This architecture was used to generate the results in the Timing-driven Titan paper: 
K. Murray et al, "Timing-Driven Titan: Enabling Large Benchmarks and Exploring the Gap between Academic and Commercial CAD," ACM TRETS, March 2015, https://dl.acm.org/doi/10.1145/2629579

File: vtr_flow/arch/titan/README.rst (hunk 1 above)
Comment (helen-23) [reply vtr_flow/arch/titan/README.rst:44]: Thank you for reviewing. I've added these comments as suggested.

File: vtr_flow/arch/titan/README.rst (hunk 2 above)
Comment (helen-23) [reply vtr_flow/arch/titan/README.rst:71]: Also added these comments as suggested.

**Pull request merged**